src/ConnectedMode/Migration/FileProviders/XmlFileProvider.cs

```diff
@@ -115,14 +143,14 @@ private IEnumerable<string> FindFiles(string rootFolder)
 
             var timer = Stopwatch.StartNew();
 
-            var allMatches = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            var allMatches = new List<string>();
```

Comment:
I moved the de-duplication using a hashset to the top-level, so we can just use a List here and call List.AddRange.